pkg/types/request.go

```diff
@@ -258,8 +259,24 @@ func ParseProviderStrings(v string) ([]peer.AddrInfo, error) {
 				return nil, err
 			}
```

Comment:
would this make sense now? seems a reasonable assumption if they're specifying the address in `http://` form

```suggestion
			}
			protocols = append(protocols, metadata.IpfsGatewayHttp{})
```

Reply:
I actually had this but then that means GetDescriptorString produces a +http which is a bit odd :P